src/huggingface_hub/utils/_errors.py

```diff
@@ -1,9 +1,67 @@
+from typing import Optional
+
 from requests import HTTPError, Response
 
+from ._deprecation import _deprecate_method
 from ._fixes import JSONDecodeError
 
 
-class RepositoryNotFoundError(HTTPError):
+class HfHubHTTPError(HTTPError):
+    """
+    HTTPError to inherit from for any custom HTTP Error raised in HF Hub.
+
+    Any HTTPError is converted at least into a `HfHubHTTPError`. If some information is
+    sent back by the server, it will be added to the error message.
+
+    Added details:
+    - Request id from "X-Request-Id" header if exists.
+    - Server error message if we can found one in the response body.
+
+    Usage:
```

Comment:
Nit: if you call say `Example:` instead of `Usage:`, there will be a link to the example that can be shared.

Reply:
Good to know, thanks !